public_html/wp-content/plugins/pattern-creator/src/plugins/category-settings-panel.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { registerPlugin } from '@wordpress/plugins';
+import { CheckboxControl } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+
+/**
+ * Internal dependencies
+ */
+import DocumentSettingPanel from '../components/document-setting-panel';
+import usePostData from '../hooks/use-post-data';
+
+/**
+ * Module constants
+ */
+const PATTERN_CAT_KEY = 'pattern-categories';
+
+const CategorySettingsPanel = () => {
+	const categories = useSelect( ( select ) =>
+		select( coreStore ).getEntityRecords( 'taxonomy', 'wporg-pattern-category', { per_page: 20 } ) || []
```

Comment:
Should we just load all categories, `-1`?

Reply:
Until we have a load more functionality we'll probably need to slice the list no matter what, otherwise it will make the Block editor pretty unusable.

Reply:
Categories is a fixed set of items though, so we shouldn't have an unusable amount.

Reply:
That's fair, but I don't know that we would ever want more than 20 categories to show up. So if the backend gets more categories (which there is a design for) we don't need to come back here in a hurry, although at that point the first page probably wouldn't be representative.

Reply:
Updated to `-1`.